pkg/operator/operator_config.go

```diff
@@ -109,24 +114,23 @@ func setupOperatorConfigControllers(op *Operator) error {
 			&monitoringv1.OperatorConfig{},
 			builder.WithPredicates(objFilterOperatorConfig),
 		).
-		// // Maintain the rule-evaluator deployment and configuration (as a secret).
 		Watches(
 			&source.Kind{Type: &appsv1.Deployment{}},
 			enqueueConst(objRequest),
 			builder.WithPredicates(
 				objFilterRuleEvaluator,
 				predicate.GenerationChangedPredicate{},
 			)).
-		// We must watch all secrets in the cluster as we copy and inline secrets referenced
-		// in the OperatorConfig and need to repeat those steps if affected secrets change.
-		// As the set of secrets to watch is not static, we've to watch them all.
-		// A viable alternative could be for the rule-evaluator (or a sidecar) to generate
-		// the configuration locally and maintain a more constrained watch.
 		Watches(
-			&source.Kind{Type: &corev1.Secret{}},
+			source.NewKindWithCache(&corev1.Secret{}, op.managedNamespacesCache),
 			enqueueConst(objRequest),
 			builder.WithPredicates(predicate.NewPredicateFuncs(secretFilter(op.opts.PublicNamespace))),
 		).
+		// Detect and undo changes to the secret.
```

Comment:
this also needs to reconcile the configmaps that the collector writes to gmp-system, such as the generated rule-evaluator config

Reply:
sorry, this controller doesn't write any configmaps. disregard